packages/compass-import-export/src/export/export-json.ts

```diff
@@ -0,0 +1,129 @@
+import { Transform } from 'stream';
+import { pipeline } from 'stream/promises';
+import { EJSON } from 'bson';
+import type { Writable } from 'stream';
+import toNS from 'mongodb-ns';
+import type { DataService } from 'mongodb-data-service';
+
+import { createReadableCollectionCursor } from '../utils/collection-stream';
+import type {
+  ExportQuery,
+  ExportAggregation,
+} from '../utils/collection-stream';
+import { createDebug } from '../utils/logger';
+
+const debug = createDebug('export-json');
+
+type ExportJSONOptions = {
+  dataService: DataService;
+  ns: string;
+  output: Writable;
+  abortSignal: AbortSignal;
+  query?: ExportQuery;
+  aggregation?: ExportAggregation;
+  progressCallback: (index: number) => void;
+  variant: 'default' | 'relaxed' | 'canonical';
+};
+
+type ExportJSONResult = {
+  docsWritten: number;
+  aborted: boolean;
+};
+
+function getEJSONOptionsForVariant(
+  variant: 'default' | 'relaxed' | 'canonical'
+) {
+  if (variant === 'relaxed') {
+    return {
+      relaxed: true,
+    };
+  }
+  return variant === 'canonical'
+    ? {
+        relaxed: false, // canonical
+      }
+    : undefined; // default
```

Comment:
Does this end up making to relaxed true/false or a third thing? Just wondering if default used to mean "compass style" ie. we could be losing some custom processing. Like a "semi-relaxed".

Reply:
I remember Anna mentioning that we do something for dates or longs or something to make the json less verbose when it doesn't need to be, but I can't find it now.